example_recursive_test.go

```diff
@@ -24,14 +26,18 @@ func recursiveServeHTTP(l net.Listener) {
 	s := &http.Server{
 		Handler: &recursiveHTTPHandler{},
 	}
-	s.Serve(l)
+	go func() {
+		if err := s.Serve(l); err != cmux.ErrListenerClosed {
+			log.Panic(err)
+		}
+	}()
 }
 
 func tlsListener(l net.Listener) net.Listener {
 	// Load certificates.
 	certificate, err := tls.LoadX509KeyPair("cert.pem", "key.pem")
 	if err != nil {
-		log.Fatal(err)
+		log.Panic(err)
```

Comment:
With the standard caveat that https://godoc.org/log#SetOutput can be used to redirect the printed error, yes